src/main/java/org/scijava/plugins/scripting/python/OptionsPython.java

```diff
@@ -208,6 +282,50 @@ public void save() {
 		}
 	}
 
+	private File writeEnvironmentYaml() {
+		File envFile = getEnvironmentYamlFile();
+
+		// skip writing if nothing has changed
+		if (initialCondaDependencies.equals(condaDependencies) &&
+			initialPipDependencies.equals(pipDependencies)) return envFile;
+
+		// Update initial dependencies to detect future changes
+		initialCondaDependencies = condaDependencies;
+		initialPipDependencies = pipDependencies;
+
+		// Write environment.yml from condaDependencies and pipDependencies
+		try {
+			String name = "fiji";
+			String[] channels = { "conda-forge" };
+			String pyimagej = "pyimagej>=1.7.0";
+			String apposePython =
+				"git+https://github.com/apposed/appose-python.git@efe6dadb2242ca45820fcbb7aeea2096f99f9cb2";
+			StringBuilder yml = new StringBuilder();
+			yml.append("name: ").append(name).append("\nchannels:\n");
+			for (String ch : channels)
+				yml.append("  - ").append(ch).append("\n");
+			yml.append("dependencies:\n");
+			for (String dep : condaDependencies.split("\n")) {
+				String trimmed = dep.trim();
+				if (!trimmed.isEmpty()) yml.append("  - ").append(trimmed).append("\n");
+			}
+			yml.append("  - pip\n");
+			yml.append("  - pip:\n");
+			for (String dep : pipDependencies.split("\n")) {
+				String trimmed = dep.trim();
+				if (!trimmed.isEmpty()) yml.append("    - ").append(trimmed).append(
+					"\n");
+			}
+			yml.append("    - ").append(pyimagej).append("\n");
```

Comment:
We should append `pyimagej>=1.7.0` only if they did not include it themselves in their list. Otherwise, it might not be possible to change the version range used, and in particular might not be possible to pin to a specific version.